fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -216,7 +216,7 @@
      */
     private List<String> dereferenceAgentGroups(final Collection<String> agentGroups) {
         // TODO do not use transactions for internal reads
-        final Transaction transaction = sessionFactory.getNewTransaction();
+        final Transaction transaction = txManager.create();
```

Comment:
reuse existing transaction?

Reply:
I will refactor this PR to no use transaction once #1561 is merged.

Reply:
I'm still unsure about this, I was under the impression that if the request was being made as part of a transaction, we would want to use the webac of the resource as it exists in the transaction rather than the head version. I've added an open question to this page so we can try to formalize a decisions:
https://wiki.duraspace.org/display/FF/Decisions+and+Open+Questions (open question 8)

Reply:
Sounds good!

Reply:
We discussed this on the tech call today and the conclusion was that all WebAC decisions should be made against the state of the WebAC as it exists within the transaction in order to be consistent with the behavior of other objects within the repository. So that would mean that we want to use the currently active transaction instead of a new one for this and the other similar cases in this PR.

Reply:
Updated to existing transaction associated with the request.